### ISSUE TITLE
Create HTTP-specific adaptation of the core spec

### ISSUE THREAD

**Issue by pkapustin** [beckn/protocol-specifications#186]
As long as the current spec is agnostic of the transport protocol, and most existing implementations are HTTP-based, we should create an HTTP-specific adaptation of the core spec. 

All HTTP implementations need to report errors correctly, and also comply with HTTP standards. Currently it is not clearly what kind of HTTP error codes need to be used in case of different errors. 

This should be standardized as much as possible in the HTTP adaptation of the core spec, rather than left to the network-specific specs.

**Comment by venkatramanm** [comment]
WE today sign only the payload and NOT the url. Many apis like init and confirm could look very similar. If tomorrow to resolve disputes, if we need to prove using signed payload that a confirm an on_confirm were indeed called. we will not be able to say for sure even if we persist the messages as some one can easily tell that it is a init json and not confirm json.  But if the action was in the context, it is included in the contract being signed. Hence we cannot  remove the action from the context!

**Comment by pkapustin** [comment]
@venkatramanm Please see my comment here.
https://github.com/beckn/protocol-specifications/issues/180#issuecomment-1013936800

However, note this issue is not about removing action from the context, but about HTTP adapation of the core spec.

**Comment by venkatramanm** [comment]
Since the 2 issues were linked, I didn't want to lose the information about action .

**Comment by pkapustin** [comment]
Yeah, I agree with you that before removing the action, even in HTTP adaptation, we have to include url in the signature.
